### ISSUE TITLE
MacOS: pycurl ImportError

### ISSUE THREAD

**Issue by schautime** [jdalbey/libbyfetch#15]
I am experiencing a pycurl error that appears related to SSL.  When I run libbyfetch.py I immediately get this error:

ImportError: pycurl: libcurl link-time ssl backends (secure-transport, openssl) do not include compile-time ssl backend (none/other)

I attempted to reinstall pycurl and open ssl shows installed and up to date.

Thanks for creating this program.  Hope I can get it to work.

**Comment by jdalbey** [comment]
Can you provide more details?  Which OS are you using?  Which version of Python?  Which version  of pyCurl?

**Comment by schautime** [comment]
Yes, sorry.  I am on a Mac Sonoma 14.6.1.  Python 3.13.0.  pyCurl 7.45.3.  I did a fresh install of python and pycurl via homebrew.  All installed successfully.  I also downgraded blinker as suggested in your instructions.  No errors on install.   But after I updated the config file for the library card data and tried to execute the script, I got that pycurl import error.  The library I am trying to connect to is the King County Library.  Lib abbreviation:  kcls

**Comment by jdalbey** [comment]
I'm sorry the program isn't working for you.  I don't have a Mac handy to investigate this.  If you would like to pursue it I can offer that when I used the error message as a search phrase for a web search several results appeared that sounded relevant.  So if you feel like diving in, try entering
`ImportError: pycurl: libcurl link-time ssl backends (secure-transport, openssl) do not include compile-time ssl backend (none/other)` into your favorite search engine. 
Sorry to toss this back to you but I think the problem is the pyCurl installation and not libbyfetch.  I know these kinds of problems are really frustrating.  If you find anything I can help with on the libbyfetch side, let me know.  If I learn anything more I'll add it in the comments.

**Comment by schautime** [comment]
Yeah, I tried to chat gpt the error before I posted here.  Chat gpt gave me some suggestions, all of which I already tried, including the reinstall of pycurl and open ssl to rebuild the appropriate connections.  

No worries, I totally understand.  Thanks for trying.  I'm going to try on my PC and see what happens.

**Comment by lsixecho** [comment]
The error:

```ImportError: pycurl: libcurl link-time ssl backends (secure-transport, openssl) do not include compile-time ssl backend (none/other)```

indicates a mismatch between the SSL backend used by the libcurl library (which pycurl depends on) and what is available on your system. pycurl expects a specific SSL backend, but your system’s libcurl library doesn’t support the required SSL backend properly at compile-time.

Why This Happens:
•The version of libcurl installed on your system may not have been compiled with an SSL backend compatible with pycurl.
•macOS often uses SecureTransport as the default SSL backend, but pycurl expects OpenSSL.

I think I used this command

```PYCURL_SSL_LIBRARY=openssl pip3 install --no-cache-dir pycurl```

to install pycurl with OpenSSL explicitly.

After that I made sure that openssl was good with:

```brew install openssl```

Then I set the correct paths for the compiler to find OpenSSL:

```export LDFLAGS="-L$(brew --prefix openssl)/lib"
export CPPFLAGS="-I$(brew --prefix openssl)/include"
export PKG_CONFIG_PATH="$(brew --prefix openssl)/lib/pkgconfig"
```

Then check if the curl version installed by Homebrew is using OpenSSL:

```curl --version```

Uninstall pycurl again and attempt to reinstall it:

```pip3 uninstall pycurl
PYCURL_SSL_LIBRARY=openssl pip3 install --no-cache-dir pycurl
```


Good luck! The dependancies are the hardest part of the libbyfetch process in my experience.

**Comment by jdalbey** [comment]
@lsixecho Thanks for these helpful instructions.  I think this explains what is causing this issue for people.  I'm sure other Mac users will appreciate your contribution.

Yes, getting the dependencies straightened out is often the crux of installation with Python applications.
